packages/swr/src/index.ts

```diff
@@ -589,23 +590,21 @@ export const ${swrMutationFetcherName} = (${swrProps} ${swrMutationFetcherOption
   }
 };
 
-export const generateSwrHeader: ClientHeaderBuilder = ({
-  isRequestOptions,
-  isMutator,
-  hasAwaitedType,
-}) =>
+export const generateSwrHeader: ClientHeaderBuilder = (params) =>
   `
   ${
-    !hasAwaitedType
+    !params.hasAwaitedType
       ? `type AwaitedInput<T> = PromiseLike<T> | T;\n
       type Awaited<O> = O extends AwaitedInput<infer T> ? T : never;\n\n`
       : ''
   }
   ${
-    isRequestOptions && isMutator
+    params.isRequestOptions && params.isMutator
       ? `type SecondParameter<T extends (...args: any) => any> = Parameters<T>[1];\n\n`
       : ''
-  }`;
+  }
+  ${params.output.httpClient === OutputHttpClient.FETCH ? generateFetchHeader(params) : ''}
```

Comment:
The functions when processing is divided by `httpClient` are consolidated into `client.ts`, so please move them 👍